reana_commons/snakemake.py

```diff
@@ -9,30 +9,41 @@
 """REANA Snakemake Workflow utils."""
 
 import os
+from itertools import filterfalse, chain
+from typing import Any, Dict, List, Optional
+
 from snakemake import snakemake
+from snakemake.dag import DAG
 from snakemake.io import load_configfile
+from snakemake.jobs import Job
+from snakemake.persistence import Persistence
+from snakemake.rules import Rule
 from snakemake.workflow import Workflow
 
 from reana_commons.errors import REANAValidationError
 
 
-def snakemake_load(workflow_file, **kwargs):
+def snakemake_load(workflow_file: str, **kwargs: Any) -> Dict:
     """Validate Snakemake workflow specification.
```

Comment:
*suggestion:* Docstring should be updated. This function is not only validating but also loading.

Reply:
*suggestion:* in general, it is not good that one function is doing two things: validating and loading (ideally, functions should do [one thing very good and nothing more](https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29#functions-rules)). I would split the code into two functions: `snakemake_validate` and `snakemake_load`.